airs2icarol/airs2icarol.py

```diff
@@ -315,133 +317,164 @@ def _translated_type_xpath(contact_type, xpath, root_element, joinstr=u'; ', _=N
+        u'PhoneFax': partial(_xpath_join, 'Phone[./Type/text() = "Fax"]/PhoneNumber/text()'),
+        u'PhoneTTY': partial(_xpath_join, 'Phone[./Type/text() = "TTY/TDD"]/PhoneNumber/text()'),
+
+        u'WebsiteAddress': partial(_xpath_join, 'URL/Address/text()'),
+        u'EmailAddressMain': partial(_xpath_join, 'Email/Address/text()'),
+
+        u'CoverageArea': partial(_xpath_join, 'GeographicAreaServed/Description/text()'),
+
+        u'MainContactTitle': partial(_translated_type_xpath, _('Primary Contact'), 'Contact[@Type = "%s"]/Title/text()'),
+        u'MainContactName': partial(_translated_type_xpath, _('Primary Contact'), 'Contact[@Type = "%s"]/Name/text()'),
+        u'MainContactPhoneNumber': partial(_translated_type_xpath, _('Primary Contact'), 'Contact[@Type = "%s"]/Phone/PhoneNumber/text()'),
+        u'MainContactEmailAddress': partial(_translated_type_xpath, _('Primary Contact'), 'Contact[@Type = "%s"]/Email/Address/text()'),
+
+        u'LastVerifiedOn': partial(_xpath_join, 'ResourceInfo/@DateLastVerified'),
+        u'LastVerifiedByName': partial(_xpath_join, 'ResourceInfo/Contact/Name/text()'),
+        u'LastVerifiedByTitle': partial(_xpath_join, 'ResourceInfo/Contact/Title/text()'),
+        u'LastVerifiedByEmailAddress': partial(_xpath_join, 'ResourceInfo/Contact/Email/Address/text()'),
+        u'LastVerifiedByPhoneNumber': partial(_xpath_join, 'ResourceInfo/Contact/Phone[Type/text() = "Voice"]/PhoneNumber/text()'),
+
+        u'FeeStructureSource': partial(_xpath_join, 'FeeStructure/text()'),
+        u'ApplicationProcess': partial(_xpath_join, 'ApplicationProcess/Description/text()'),
+        u'LanguagesOffered': _languages_offered,
+        u'InternalNotesForEditorsAndViewers': partial(_xpath_join, 'InternalNote/text()'),
+        u'InternalNote': partial(_xpath_join, 'EditorsNote/text()'),
+        u'Custom_PublicComments': partial(_xpath_join, 'PublicNote/text()'),
+        u'Eligibility': _eligibility,
+        u'HoursOfOperation': partial(_xpath_join, 'TimeOpen/Notes/text()'),
+        u'Custom_FormerNames': partial(_xpath_join, 'AKA/Name[starts-with(text(),"FORMER")]/text()'),
+        u'Custom_LegalNames': partial(_xpath_join, 'AKA/Name[starts-with(text(),"LEGAL")]/text()'),
+
+    }
 
 
 }
+_header_prefix = ['AgencyNUM', 'NUM', 'Record Type', 'CultureCode']
+
+# Get all keys from dictionary
+_field_order = [key for entry in _airs_type_mapping.values() for key in entry.keys()]
+
+# Remove duplicates
+_field_order = sorted(list(set(_field_order)))
+
+# Remove the memo items and place them at the end of the list
+###This could instead be done with 3 calls to remove followed by 3 calls to append but this is easier to maintain if a new memo is added
+_last_items = ['InternalNotesForEditorsAndViewers','InternalNote','Custom_PublicComments','Custom_FormerNames','Custom_LegalNames']
```

Comment:
Only Custom_* fields are stupposed to be at the end. The other ones are normal fields and shouldn't be forced to the end.

There is a more efficient approach to sticking the items at the end by doing using a sort function as the key for ```sorted()```

move up _last_items and then create your sort function e.g.

``` py
def sort_fn(value):
     return (value in _last_items, value)
```